parachain/pallets/inbound-queue/src/test.rs

```diff
@@ -330,53 +41,20 @@ fn test_submit_happy_path() {
 #[test]
 fn test_submit_xcm_invalid_channel() {
 	new_tester().execute_with(|| {
-		let relayer: AccountId = Keyring::Bob.into();
-		let origin = RuntimeOrigin::signed(relayer);
-
-		// Deposit funds into sovereign account of parachain 1001
-		let sovereign_account = sibling_sovereign_account::<Test>(1001u32.into());
-		println!("account: {}", sovereign_account);
-		let _ = Balances::mint_into(&sovereign_account, 10000);
-
-		// Submit message
-		let message = Message {
-			event_log: mock_event_log_invalid_channel(),
-			proof: Proof {
-				block_hash: Default::default(),
-				tx_index: Default::default(),
-				data: Default::default(),
-			},
-		};
-		assert_noop!(
-			InboundQueue::submit(origin.clone(), message.clone()),
-			Error::<Test>::InvalidChannel,
-		);
+		let log = mock_event_log_invalid_channel();
+		let envelope = InboundQueue::decode(log).unwrap();
+		let err = InboundQueue::validate(&envelope).unwrap_err();
+		assert_eq!(err, Error::<Test>::InvalidChannel);
 	});
 }
 
 #[test]
 fn test_submit_with_invalid_gateway() {
 	new_tester().execute_with(|| {
-		let relayer: AccountId = Keyring::Bob.into();
```

Comment:
What was wrong with this approach to testing? I felt it more complete because it was calling `submit` which is what the user will call.

Now its calling `decode` and `validate` instead, and I can't the benefits to this approach.